### DISCUSSION TITLE
How do I manually add a queue to topic to a AMQ session?

### DISCUSSION THREAD

**Discussion by ErnestoRuiz** [jmstoolbox/jmstoolbox#157] [Q&A] [answered]
I've got JMSToolBox connected to a remote AMQ broker and it's currently displaying the queues it's discovered. However, there are many other queues that exist. Is there a way to publish a message to a queue or topic that I've manually set.

**Comment by titou10titou10** [comment]
There is no way to manually add a queue 
What do you mean by "here are many other queues that exist." ? Can you see thoses queue from the official ActiveMQ console with the same userid you are using on JTB? can you post message via the console to those queues?
Do you filter the queues you are seeing in the JTB queue browser? are those queue temporary or system queues? are they "JMS" queues? What do those "not visible queues" not visible in JTB have in common?

**Reply by ErnestoRuiz** [reply]
So the remote ActiveMq admin console shows there to be dozens of queues and topics that used by other applications. But when I connect to them using JTB, I only see four queues and four topics. 

I am using the same credentials on JTB as I am to log in to the remote AMQ console. I also am using the same connection URL as I did when I used Hermes JMS and I was then ABLE to send messages to queues and topics without issue.

I used to use Hermes JMS, but since I switch to a Mac, I have to use something else because Hermes/SoapUI does not run on the ARM64 Macs. 

The reason I’m bringing up Hermes is because I was able to right click on the session that I created and add queue or topic then just send a message to the queue or topic I’ve added.

I thought there might be a similar function that I was overlooking. My apologies for having expectations based on my experience with different piece of software.

**Comment by titou10titou10** [comment]
But what make those 8 Q/Topics different from the other Q/Topics?
Are there specific messages in the JTB logs?
In the preference, did you select to show temp/system destinations?

**Reply by ErnestoRuiz** [reply, marked as answer]
Okay, I’ve figured it out. I just had to rescan destinations and now I see all the queues and topics now. Sorry to be a bother.
